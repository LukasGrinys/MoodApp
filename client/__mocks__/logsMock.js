export const testLogs = [
    {
        id: '001',
        rating: 10,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '002',
        rating: 9,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '003',
        rating: 8,
        date: "TEST-DATE",
        text: "Test text"
    },
    {
        id: '004',
        rating: 7,
        date: "TEST-DATE",
        text: "Test text"
    }
];

export const testLog = {
    id: '001',
    rating: 10,
    timing: 'Afternoon', 
    date: "TEST-DATE",
    text: "Test text"
}