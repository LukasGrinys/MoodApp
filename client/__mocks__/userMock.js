export const testUser = {
    id: '001',
    firstName: 'test',
    lastName: 'test',
    email: 'test@test.com'
}