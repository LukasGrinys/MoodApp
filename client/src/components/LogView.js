import React, { Component } from 'react';
import { viewLog, clearLog } from './../actions';
import { connect } from 'react-redux';
import Loading from './../widgets/loading';
import styles from './logview.module.css';
import FontAwesome from 'react-fontawesome';
import BackButton from './../widgets/backButton';

class LogView extends Component {
    componentWillMount() {
        this.props.dispatch(viewLog(this.props.match.params.id))
    }
    componentWillUnmount() {
        this.props.dispatch(clearLog())
    }
    render() {
        if (this.props.logs) {
            if (this.props.logs.logInfo) {
                const log = this.props.logs.logInfo.log
                if (log) {
                    return (
                        <div>
                            <div className={styles.top_panel}>
                            <BackButton nightmode={this.props.nightmode}/>
                            </div>
                            <div className={styles.logWrapper}>
                                <div className={styles.top_line}>
                                    <div className={styles.rating_box}>
                                        {log.rating}
                                    </div>
                                    <div>
                                        <FontAwesome name="hourglass" className={styles.icon}/>
                                        {log.timing}
                                    </div>
                                    <div className={styles.date}>
                                        <FontAwesome name="calendar" className={styles.icon}/>
                                        {log.date}
                                    </div>
                                </div>
                                <div className={styles.text}>
                                    {log.text}
                                </div>
                            </div>
                        </div>
                    );
                } else {
                    return <Loading nightmode={this.props.nightmode}/>
                }
                
            } else {
                return <Loading nightmode={this.props.nightmode}/>
            }     
        } else {
            return <Loading/>
        } 
    }
}

function mapStateToProps(state) {
    return {
        logs: state.logs
    }
}

export default connect(mapStateToProps)(LogView);