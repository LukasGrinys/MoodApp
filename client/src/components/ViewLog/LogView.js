import React, { useEffect } from 'react';
import { getLog, clearLog } from '../../actions/logs/asyncActions';
import { useDispatch, useSelector } from 'react-redux';
import Loading from '../Loading/loading';
import BackButton from '../Back';
import LogItem from '../LogItem/logItem';
import ErrorBox from '../ErrorBox';
import styles from './logview.module.css';

const LogView = (props) => {
    const dispatch = useDispatch();
    const { logData, isFetching : isFetchingLog, error : getLogError } = useSelector( ({logs}) => logs.singleLog);

    useEffect( () => {
        if (
            props.match && 
            props.match.params &&
            props.match.params.id
        ) {
            dispatch(getLog(props.match.params.id));
        }

        return () => {
            dispatch(clearLog());
        }
        // eslint-disable-next-line
    }, []);

    if (!logData || isFetchingLog) {
        return <Loading/>
    }

    return (
        <div className={styles.pageWrapper}>
            <BackButton/>
            {
                logData && Boolean(Object.keys(logData).length) && (
                    <LogItem 
                        rating={logData.rating}
                        timing={logData.timing}
                        date={logData.date}
                        text={logData.text}
                    />
                )
            }
            {
                getLogError && (
                    <ErrorBox>
                        {getLogError}
                    </ErrorBox>
                )
            }
        </div>
    )
}

export default LogView