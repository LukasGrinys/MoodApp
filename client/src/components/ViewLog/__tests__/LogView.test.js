import React from 'react';
import LogView from '../LogView';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { testLog } from '../../../../__mocks__/logsMock';

const mockStore = configureStore();
const mockDispatch = jest.fn();

jest.mock("react-redux", () => {
    const { Provider, useSelector } = jest.requireActual("react-redux");
  
    return {
        useDispatch: () => mockDispatch,
        useSelector,
        Provider
    };
});

describe('Initial LogView', () => {
    const store = mockStore({
        user: {},
        logs: {
            singleLog: {}
        }
    })
    
    const testId = 'testId';
    const initialProps = {
        match: {
            params: {
                id: testId
            }
        }
    }
    
    const wrapper = mount(
        <Provider store={store}>
            <LogView {...initialProps}/>
        </Provider>
    );
    
    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Shows loading widget', () => {
        expect(wrapper.find('Loading').length).toBe(1);
    });

    test('Dispatches getLog action', () => {
        setTimeout( () => {
            const getLog = jest.fn();

            expect(getLog).toHaveBeenCalledTimes(1);
            expect(getLog).toHaveBeenCalledWith(testId);

            done();
        }, 0);
    })
});

describe('LogView after log data was fetched', () => {
    const store = mockStore({
        user: {},
        logs: {
            singleLog: {
                logData: {
                    ...testLog,
                },
                isFetching: false,
                error: null
            }
        }
    })
    
    const wrapper = mount(
        <Provider store={store}>
            <LogView/>
        </Provider>
    );

    test('Renders properly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Renders LogItem with proper props', () => {
        expect(wrapper.find('LogItem').length).toBe(1);
        expect(wrapper.find('LogItem').props()).toEqual({
            rating: testLog.rating,
            timing: testLog.timing,
            date: testLog.date,
            text: testLog.text
        })
    })
});

test('Renders error if there was an error while fetching log', () => {
    const testError = 'Test error';

    const store = mockStore({
        user: {},
        logs: {
            singleLog: {
                logData: {},
                isFetching: false,
                error: testError
            }
        }
    })
    
    const wrapper = mount(
        <Provider store={store}>
            <LogView/>
        </Provider>
    );

    expect(wrapper.find('ErrorBox').length).toBe(1);
    expect(wrapper.find('ErrorBox').text()).toBe(testError);
})
