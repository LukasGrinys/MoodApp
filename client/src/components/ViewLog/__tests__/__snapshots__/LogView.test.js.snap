// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial LogView Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LogView
    match={
      Object {
        "params": Object {
          "id": "testId",
        },
      }
    }
  >
    <Loading>
      <div
        className="loadingScreen"
      >
        <div>
          Loading
        </div>
        <br />
        <div
          className="barContainer"
        >
          <div
            className="bar"
          />
          <div
            className="bar delay1"
          />
          <div
            className="bar delay2"
          />
        </div>
      </div>
    </Loading>
  </LogView>
</Provider>
`;

exports[`LogView after log data was fetched Renders properly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <LogView>
    <div
      className="pageWrapper"
    >
      <BackButton>
        <div
          className="container"
        >
          <div
            className="button"
            onClick={[Function]}
          >
            <FontAwesome
              name="arrow-left"
            >
              <span
                aria-hidden={true}
                className="fa fa-arrow-left"
              />
            </FontAwesome>
            <div
              className="text"
            >
              Go back
            </div>
          </div>
        </div>
      </BackButton>
      <LogItem
        date="TEST-DATE"
        rating={10}
        text="Test text"
        timing="Afternoon"
      >
        <div
          className="item"
        >
          <div
            className="top_line"
          >
            <div
              className="rating_box"
            >
              10
            </div>
            <div>
              <FontAwesome
                className="icon"
                name="hourglass"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-hourglass icon"
                />
              </FontAwesome>
              Afternoon
            </div>
            <div
              className="date"
            >
              <FontAwesome
                className="icon"
                name="calendar"
              >
                <span
                  aria-hidden={true}
                  className="fa fa-calendar icon"
                />
              </FontAwesome>
              TEST-DATE
            </div>
          </div>
          <div
            className="text"
          >
            Test text
          </div>
        </div>
      </LogItem>
    </div>
  </LogView>
</Provider>
`;
